pyodide-build/pyodide_build/buildpkg.py

```diff
@@ -784,27 +788,29 @@ def build_package(
         run_script(build_dir, srcpath, build_metadata, bash_runner)
 
         if library:
-            create_packaged_token(build_dir)
-            return
-
-        if not sharedlibrary and not finished_wheel:
-            compile(
-                name,
-                srcpath,
-                build_metadata,
-                bash_runner,
-                target_install_dir=target_install_dir,
-            )
-        if not sharedlibrary:
+            # Nothing needs to be done for a static library
+            pass
+        elif sharedlibrary:
+            # If shared library, we copy .so files to dist_dir
+            # and create a zip archive of the .so files
+            shutil.rmtree(dist_dir, ignore_errors=True)
+            dist_dir.mkdir(parents=True)
+            shutil.make_archive(str(dist_dir / src_dir_name), "zip", srcpath / "dist")
```

Comment:
It's a bit hard to keep track of all these similar combinations of things...
```suggestion
            shutil.make_archive(str(dist_dir / src_dir_name), "zip", src_dist_dir)
```

Reply:
Oh, thanks. I forgot this one.